backend/tests/lib.rs

```diff
@@ -1163,76 +1169,378 @@ fn test_content() {
+    let response = client
+        .delete(format!("/burrows/{}", burrow_id))
+        .remote("127.0.0.1:8000".parse().unwrap())
+        .dispatch();
+    assert_eq!(response.status(), Status::Ok);
+    let res = response.into_string().unwrap();
+    assert_eq!(res, format!("Success"));
+
+    //retrieve a discarded burrow
+    let response = client
+        .post(format!("/search/?{}", 1))
+        .json(&SearchRequest::RetrieveBurrow {
+            burrow_id: burrow_id,
+        })
+        .remote("127.0.0.1:8000".parse().unwrap())
+        .dispatch();
+    assert_eq!(response.status(), Status::Ok);
+    let res = response.into_json::<serde_json::Value>().unwrap();
+    assert_eq!(res["title"], "Default".to_string());
+    // println!("Retrieve result: {}", response.into_string().unwrap());
+
+    //retrieve a non-exist post
+    let response = client
+        .post(format!("/search/?{}", 1))
+        .json(&SearchRequest::RetrievePost { post_id: -1 })
+        .remote("127.0.0.1:8000".parse().unwrap())
+        .dispatch();
+    assert_eq!(response.status(), Status::NotFound);
+    let res = response.into_json::<ErrorResponse>().unwrap();
+    // println!("Retrieve result: {}", response.into_string().unwrap());
+    assert_eq!(res.error.code, ErrorCode::PostNotExist);
+    assert_eq!(res.error.message, "Cannot find post -1".to_string());
+}
+
+// #[test]
+fn test_storage() {
+    let client = common::get_client().lock();
+    // generate a random name
+    let name: String = std::iter::repeat(())
+        .map(|()| thread_rng().sample(Alphanumeric))
+        .map(char::from)
+        .take(16)
+        .collect();
+
+    // sign up a user
+    let response = client
+        .post("/users/sign-up")
+        .json(&json!({
+        "username": name,
+        "password": "testpassword",
+        "email": format!("{}@mails.tsinghua.edu.cn", name)}))
+        .remote("127.0.0.1:8000".parse().unwrap())
+        .dispatch();
+    assert_eq!(response.status(), Status::Ok);
+    let res = response
+        .into_json::<backend::models::user::UserResponse>()
+        .unwrap();
+    let burrow_id = res.default_burrow;
+
+    // user login
+    let response = client
+        .post("/users/login")
+        .json(&json!({
+            "username": name,
+            "password": "testpassword"}))
+        .remote("127.0.0.1:8000".parse().unwrap())
+        .dispatch();
+    assert_eq!(response.status(), Status::Ok);
+    // println!("{}", response.into_string().unwrap());
+    std::thread::sleep(std::time::Duration::from_secs(5));
+
+    //get an jepg from httpbin
+    let mut jpeg_buf: Vec<u8> = vec![];
+    match reqwest::blocking::Client::new()
+        .get("http://httpbin.org/image/jpeg")
+        .send()
+    {
+        Ok(mut r) => match r.status() {
+            StatusCode::OK => {
+                r.copy_to(&mut jpeg_buf).unwrap();
+            }
+            _ => {
+                let jpeg: String = std::iter::repeat(())
```

Comment:
Create the jpeg string outside of the match, so that you can use it when match a Err.